src/System.Text.Json/docs/writable_json_dom_spec.md

```diff
@@ -0,0 +1,253 @@
+* Documentation for public API
+
+## Example scenarios
+### Collection initialization
+
+One of the aims in designing this API was the take an advantage of C# language features and make it easy and natural for delevopers to create instances of `JsonObject`s without calling too many `new` instructions. Below example shows how to initialize JSON object with different types of properties:
+
+```csharp
+var developer = new JsonObject
+{
+    { "name", "Kasia" },
+    { "age", 22 },
+    { "is developer", true },
+    { "null property", (JsonNode) null }
+};
+```
+
+JSON object can be nested within other JSON object or include a JSON array: 
+
+```csharp
+var person = new JsonObject
+{
+    { "name", "John" },
+    { "surname", "Smith" },
+    {
+        "addresses", new JsonObject()
+        {
+            {
+                "office", new JsonObject()
+                {
+                    {  "address line 1", "One Microsoft Way" },
+                    {  "city" , "Redmond" } ,
+                    {  "zip code" , 98052 } ,
+                    {  "state" , (int) AvailableStateCodes.WA }
+                }
+            },
+            {
+                "home", new JsonObject()
+                {
+                    {  "address line 1", "Pear Ave" },
+                    {  "address line 2", "1288" },
+                    {  "city" , "Mountain View" } ,
+                    {  "zip code" , 94043 } ,
+                    {  "state" , (int) AvailableStateCodes.CA }
+                }
+            }
+        }
+    },
+    {
+        "phone numbers", new JsonArray()
+        {
+            "123-456-7890",
+            "123-456-7890" 
+        }
+    }
+};
+```
+
+JSON array can be also initialized easily in various ways which might be useful in different scenarios:
+
+```csharp
+string[] dishes = { "sushi", "pasta", "cucumber soup" };
+IEnumerable<string> sports = sportsExperienceYears.Where(sport => ((JsonNumber)sport.Value).GetInt32() > 2).Select(sport => sport.Key);
+
+var preferences = new JsonObject()
+{
+    { "colours", new JsonArray { "red", "green", "purple" } },
+    { "numbers", new JsonArray { 4, 123, 88 } },
+    { "prime numbers", new JsonNumber[] { 19, 37 } },
+    { "dishes", new JsonArray(dishes) },
+    { "sports", new JsonArray(sports) },
+    { "strange words", strangeWords.Where(word => ((JsonString)word).Value.Length < 10) },
+};
+```
+
+### Modifying existing instance
+
+The main goal of the new API is to allow users to modify existing instance of `JsonNode` which is not possible with `JsonElement` and `JsonDocument`. 
+
+One may change the existing property to have a different value:
+```csharp
+ var options = new JsonObject { { "use caching", true } };
+ options["use caching"] = (JsonBoolean)false;
+```
+
+Add a value to existing JSON array or property to existing JSON object:
+```csharp
+var bestEmployees = new JsonObject(EmployeesDatabase.GetTenBestEmployees());
+bestEmployees.Add(EmployeesDatabase.GetManager());
```

Comment:
Does it make sense to add an object (like  `EmployeesDatabase.GetManager()`) to a JsonObject without a property name, or should sample be about creating a JsonArray?

Reply:
Right, thanks for noticing this. In my examples (https://github.com/kasiabulat/corefx/blob/master/src/System.Text.Json/tests/WritableJsonApiTests.TestModifying.cs) `GetTenBestEmployees` was returning `IEnumerable<KeyValuePair<string, JsonNode>>`, but `GetManager()` actually returned `JsonObject`, so this scenario mixes two things and doesn't work. I can create two separate, clear one examples.